src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -63,7 +63,8 @@ public static ConsoleColor BackgroundColor
 
         public static void ResetColor()
         {
-            if (!ConsoleOutIsTerminal)
+            // We only want to reset colors if we're targeting a TTY device
+            if (Console.IsOutputRedirected)
```

Comment:
I don't think this is correct.  IsOutputRedirected is caching the result, but that will cause problems here if someone replaces the output stream with Console.SetOut.